src/server/game/Entities/Item/ItemTemplate.cpp

```diff
@@ -129,6 +129,34 @@ uint32 ItemTemplate::GetSkill() const
     }
 }
 
+bool ItemTemplate::IsWeaponOfSubClass(ItemSubclassWeapon subClass) const
+{
+    if (Class != ITEM_CLASS_WEAPON)
+        return false;
+
+    return SubClass == subClass;
+}
+
+bool ItemTemplate::IsWeaponOfSubClass(std::initializer_list<ItemSubclassWeapon> subClasses) const
+{
+    if (Class != ITEM_CLASS_WEAPON)
+        return false;
+
+    for (const auto& subClass : subClasses)
+        if (IsWeaponOfSubClass(subClass))
+            return true;
+
+    return false;
+}
+
+bool ItemTemplate::CanBeDualWieldedWithTitansGrip() const
+{
```

Comment:
Titan's Grip already has SubclassMask defined, maybe use that?

Reply:
Not sure what you think here, little help ?

Reply:
Titan's Grip spell:

```
ID - 46917 Titan's Grip (Rank 1)
=================================================
Description: Allows you to equip two-handed axes, maces and swords in one hand.  While you have a two-handed weapon equipped in one hand, your physical damage done is reduced by $49152s1%.
Category = 0, SpellIconID = 2750, activeIconID = 0, SpellVisual = (0,0)
Family SPELLFAMILY_WARRIOR, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x000001C0 (SPELL_ATTR0_PASSIVE, SPELL_ATTR0_HIDDEN_CLIENTSIDE, SPELL_ATTR0_HIDE_IN_COMBAT_LOG)
=================================================
Skill (Id 256) "Fury"
    ReqSkillValue 1, Forward Spell = 0, MinMaxValue (0, 0), CharacterPoints (0, 0)Spell Level = 0, base 0, max 0, maxTarget 0
EquippedItemClass = 2 (WEAPON)
    SubClass mask 0x0000A1B3 (AXE, AXE2, MACE, MACE2, SWORD, SWORD2, FIST, DAGGER)
    InventoryType mask = 0x00600000 (WEAPONMAINHAND, WEAPONOFFHAND)
```

Reply:
I see now, thank you

Reply:
something like:

```cpp
bool ItemTemplate::CanBeDualWieldedWithTitansGrip() const
{
    SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(46917); // Titan Grip
    return ((1 << SubClass) & spellInfo->EquippedItemSubClassMask) != 0;
}
```